### PR TITLE
#159987417 Users should be able to bookmark articles for reading later

### PR DESCRIPTION
**Pull request by joeeasy**
**What does this PR do?**
Implements a feature that allows users to: 
- bookmark an article
- view all their bookmarked articles

**Description of Task to be completed?**
- write the test for bookmarking articles
- write the test for getting all bookmarked articles
- write documentation for bookmarking an article
**How should this be manually tested?**
Clone the repository.
Check out the `ft-users-can-bookmark-article` branch.
Run `NPM INSTALL` on your local machine.
Run database migrations using `npm run migrate`.
Start the server using `NPM RUN START_DEV`.

**You can test the following endpoints using postman**
POST `localhost:8000/api/v1/articles/bookmarks` to bookmark an article
GET `localhost:8000/api/v1/articles/bookmarks` to view all bookmarked articles
HEADERS
- authorization -> paste your token there

REQUIRED FIELDS
POST
-    title: `string`
-   id(article ID): `number`

**What are the relevant pivotal tracker stories?**
#159987417

### CODE REVIEW

File: server/controllers/articleController.js [hunk 1]
```diff
@@ -66,6 +68,59 @@ const articleController = {
...
+   * @description allows users to book mark an article
+   * @param {Object} req the response object
+   * @param {Object} res the response object
+   * @returns {res} the booked marked articles
+   */
+  createBookmark: (req, res) => {
+    const { title, id } = req.body;
+    const articleId = parsedId(id);
+    if (!Number.isInteger(articleId)) {
+      return res.status(500).jsend.fail({
```
Comment (IyiKuyoro) [inline server/controllers/articleController.js]: This is not a server error

Comment (joeeasy) [reply server/controllers/articleController.js]: Thanks, man!
I've quickly fixed it

File: server/controllers/articleController.js [hunk 2]
```diff
@@ -106,6 +108,61 @@ const articlesController = {
...
+      });
+    }
+    Bookmarks.findOrCreate({
+      where: {
+        title,
+        userId: req.currentUser.id,
+        articleId,
+      }
+    })
+      .spread((bookmark, created) => (!(created) ? res.status(400).jsend.fail({ message: 'You have already bookmarked this article' })
```
Comment (jojitoon) [inline server/controllers/articleController.js]: good. but for easy readability put the ternary operator in this format: 
```
condition to check
? value if the condition is satisfied
: value if the condition is not satisfied
```

**Review by jojitoon — CHANGES_REQUESTED** [review]

File: server/controllers/articleController.js [hunk 3]
```diff
@@ -264,7 +267,64 @@ const articlesController = {
...
+        title,
+        userId: req.currentUser.id,
+        articleId,
+      }
+    })
+      .spread((bookmark, created) => (!(created)
+        ? res.status(400).jsend.fail({ message: 'You have already bookmarked this article' })
+        : res.status(201).jsend.success({
+          bookmark,
+          message: 'Your book mark has been created successfully'
```
Comment (iAmao) [inline server/controllers/articleController.js]: `bookmark` not `book mark`

File: server/controllers/articleController.js [hunk 4]
```diff
@@ -264,7 +267,64 @@ const articlesController = {
...
+      where: {
+        userId: id
+      }
+    })
+      .then(bookmarks => ((bookmarks.length === 0)
+        ? res.status(404).jsend.fail({
+          message: 'You have not bookmarked any article'
+        })
+        : res.status(200).jsend.success({
+          message: 'Here are your bookmarks',
```
Comment (iAmao) [inline server/controllers/articleController.js]: We do not need a message here

**Review by iAmao — CHANGES_REQUESTED** [review]

**Pull request merged**